### ISSUE TITLE
Unable to use ktlint 0.38.0-alpha01 (Kotlin 1.4 compatibility)

### ISSUE THREAD

**Issue by ephemient** [jeremymailen/kotlinter-gradle#152]
```kotlin
buildscript {
    configurations.classpath.resolutionStrategy.dependencySubstitution.all {
        val requested = requested as? ModuleComponentSelector ?: return@all
        if (requested.group == "com.pinterest.ktlint") {
            useTarget("${requested.moduleIdentifier}:0.38.0-alpha01", "Kotlin 1.4 compatibility")
        }
    }
}
```
```
$ ./gradlew lintKotlin
Execution failed for task ':lintKotlin'.
> There was a failure while executing work items
   > A failure occurred while executing org.jmailen.gradle.kotlinter.tasks.lint.LintWorkerRunnable
      > kotlin.jvm.internal.FunctionReferenceImpl.<init>(ILjava/lang/Object;Ljava/lang/Class;Ljava/lang/String;Ljava/lang/String;I)V
```

Kotlinter itself has compatibility problems with Kotlin 1.4?

**Comment by jeremymailen** [comment]
Probably another manifestation of this: https://github.com/pinterest/ktlint/issues/830#issuecomment-675342084
I'm watching the threads, but haven't worked through the upgrade yet. It may be that there are still some incompatibilities in alpha01.

**Comment by ephemient** [comment]
Experimenting with some workarounds, I think this might actually require #58. We're trying to use Kotlin 1.4 compiled classes, but because it's mixed with Gradle classpath containing Kotlin 1.3 stdlib, we end up with incompatibilities.

**Comment by jeremymailen** [comment]
Yeah, I tried the upgrade this evening and the issue is that if you are using the Gradle Kotlin DSL, that implementation is still pinned to kotlin 1.3 and causes a clash with this plugin would need 1.4.

I guess the options are to wait for a version of gradle which upgrades to kotlin 1.4 or implement isolation of kotlinter's classloader.

**Comment by henrik242** [comment]
0.38 is released: https://github.com/pinterest/ktlint/pull/846

**Comment by jeremymailen** [comment]
3.0.0 is now available with kotlin 1.4.0 support.

**Comment by ssp** [comment]
Thanks for the update @jeremymailen !
